packages/legend-query/src/stores/QueryBuilderState.ts

```diff
@@ -242,7 +245,9 @@ export class QueryBuilderState {
       this.buildStateFromRawLambda(rawLambda);
     } catch (error) {
       assertErrorThrown(error);
+      const queryParametersState = this.queryParametersState;
       this.changeClass(undefined, true);
+      this.queryParametersState = queryParametersState;
       if (options?.notifyError) {
         this.applicationStore.notifyError(
           `Unable to initialize query builder: ${error.message}`,
```

Comment:
I would leave everything else the same and handle the building of the parameter value spec and parameter state here, levearging `processQueryParameters`. 
You can build a LambdaFunction using a RawLambda with parameters and an empty body. 
You can then use theat Lambda Function's variable expressions to build the query parameter state.